apidef/oas/default.go

```diff
@@ -73,19 +73,21 @@ func (s *OAS) BuildDefaultTykExtension(overRideValues TykExtensionConfigParams)
 
 	if overRideValues.UpstreamURL != "" {
 		upstreamURL = overRideValues.UpstreamURL
-	} else {
+	} else if xTykAPIGateway.Upstream.URL == "" {
```

Comment:
why else if instead of else ?

Reply:
this is an edge case. 
so we need to use `servers` object only during import ie when `xTykAPIGateway.Upstream.URL` is empty. 
else if user wants to patch using some other params **only**, say `listenPath`, at that time it'll try to rely on servers object - which could be empty, or it could be something which we added, moreover there's a chance that it might override whatever upstream URL the user already configured with